src/Microsoft.Health.SqlServer/ISqlConnectionFactory.cs

```diff
@@ -0,0 +1,18 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Data.SqlClient;
+
+namespace Microsoft.Health.SqlServer
+{
+    public interface ISqlConnectionFactory
```

Comment:
Factory class is not the iterface, it is ISqlConnection or IHealthSqlConnection. 
with a method to GetSqlConnection(). Then you would have implementations like ManagedIdentitySqlConnection: ISqlConnection,  SqlUserNamePasswordSqlConnection: ISqlConnection. And then you would have a SqlConnectionFactory that would return the right implementation of ISqlConnection

Reply:
Updated as per our discussion offline.